src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -502,6 +506,19 @@ def python_type_to_snow_type(tp: Union[str, Type]) -> Tuple[DataType, bool]:
     if tp == Geometry:
         return GeometryType(), False
 
+    if tp == Timestamp or tp_origin == Timestamp:
+        if not tp_args:
+            timezone = TimestampTimeZone.DEFAULT
+        elif tp_args[0] == NTZ:
+            timezone = TimestampTimeZone.NTZ
+        elif tp_args[0] == LTZ:
+            timezone = TimestampTimeZone.LTZ
+        elif tp_args[0] == TZ:
+            timezone = TimestampTimeZone.TZ
+        else:
```

Comment:
Just curious: what can this `else` be? Or should we actually raise?

Reply:
Actually it can be `Timestamp[int]`, which is invalid and we should raise an exception here, good catch! I added tests to cover these code path.